pkg/interop/convert/convert.go

```diff
@@ -20,3 +20,247 @@ func ToString(v any) string {
+		res[i] = byte(n & 0xff)
+		n = n >> 8
+	}
+	return res
+}
+
+// Uint64ToBEBytes converts an uint64 to a fixed-length slice in big-endian format.
+func Uint64ToBEBytes(n uint64) []byte {
+	return UintToBEBytes(n, 8)
+}
+
+// Uint32ToBEBytes converts an uint32 to a fixed-length slice in big-endian format.
+func Uint32ToBEBytes(n uint32) []byte {
+	return UintToBEBytes(uint64(n), 4)
+}
+
+// Uint16ToBEBytes converts an uint16 to a fixed-length slice in big-endian format.
+func Uint16ToBEBytes(n uint16) []byte {
+	return UintToBEBytes(uint64(n), 2)
+}
+
+// Uint8ToBEBytes converts an uint8 to a fixed-length slice in big-endian format.
+func Uint8ToBEBytes(n uint8) []byte {
+	return UintToBEBytes(uint64(n), 1)
+}
+
+// IntToBEBytes converts an int64 to a slice of a specified length
+// in big-endian format.
+func IntToBEBytes(m int64, width uint64) []byte {
+	// without creating a new variable, compilation fails with panic, check (#3993)
+	var n = m
+	if width == 0 {
+		return []byte{}
+	}
+	n = n + (1 << (width*8 - 1))
+	return UintToBEBytes(uint64(n), width)
+}
+
+// Int64ToBEBytes converts an int64 to a fixed-length slice in big-endian format.
+func Int64ToBEBytes(n int64) []byte {
+	return IntToBEBytes(n, 8)
+}
+
+// Int32ToBEBytes converts an int32 to a fixed-length slice in big-endian format.
+func Int32ToBEBytes(n int32) []byte {
+	return IntToBEBytes(int64(n), 4)
+}
+
+// Int16ToBEBytes converts an int16 to a fixed-length slice in big-endian format.
+func Int16ToBEBytes(n int16) []byte {
+	return IntToBEBytes(int64(n), 2)
+}
+
+// Int8ToBEBytes converts an int8 to a fixed-length slice in big-endian format.
+func Int8ToBEBytes(n int8) []byte {
+	return IntToBEBytes(int64(n), 1)
+}
+
+// BEBytesToUint64 converts data in big-endian format to an uint64.
+func BEBytesToUint64(data []byte) uint64 {
+	if len(data) != 8 {
+		panic("expected data with length 8")
+	}
+	//nolint:govet,staticcheck
+	return uint64(data[7] | data[6]<<8 | data[5]<<16 | data[4]<<24 |
+		data[3]<<32 | data[2]<<40 | data[1]<<48 | data[0]<<56)
+}
+
+// BEBytesToUint32 converts data in big-endian format to an uint32.
+func BEBytesToUint32(data []byte) uint32 {
+	if len(data) != 4 {
+		panic("expected data with length 4")
+	}
+	//nolint:govet,staticcheck
+	return uint32(data[3] | data[2]<<8 | data[1]<<16 | data[0]<<24)
+}
+
+// BEBytesToUint16 converts data in big-endian format to an uint16.
+func BEBytesToUint16(data []byte) uint16 {
+	if len(data) != 2 {
+		panic("expected data with length 2")
+	}
+	//nolint:govet,staticcheck
+	return uint16(data[1] | data[0]<<8)
+}
+
+// BEBytesToUint8 converts data in big-endian format to an uint8.
+func BEBytesToUint8(data []byte) uint8 {
+	if len(data) != 1 {
+		panic("expected data with length 1")
+	}
+	return data[0]
+}
+
+// BEBytesToInt64 converts data in big-endian format to an int64.
+func BEBytesToInt64(data []byte) int64 {
+	//nolint:all // something reason
+	return int64(BEBytesToUint64(data) - (uint64(1) << 63))
```

Comment:
`(uint64(1) << 63)` may be moved to a separate constant.

@roman-khimov, do we need `int*` helpers, or we're OK with just `uint*` set?